packages/neon_framework/lib/src/utils/request_manager.dart

```diff
@@ -26,7 +28,7 @@ typedef SerializeCallback<T> = String Function(T);
 
 /// A callback to revive cached values [String] into their original type [T].
 @internal
-typedef DeserializeCallback<T> = T Function(String);
+typedef DeserializeCallback<T> = FutureOr<T> Function(String);
```

Comment:
Let's keep this synchronous
```suggestion
typedef DeserializeCallback<T> = T Function(String);
```